content/docs/grammar-language.md

```diff
@@ -0,0 +1,371 @@
+```
+Person:
+    'person' name=ID;
+Greeting:
+    'Hello' person=[Person:ID] '!';
+```
+The `Person` in square brackets does not refer to the parser rule `Person` but instead refers to an object of type `Person`. It will successfully parse a document like:
+```
+person Bob
+Hello Bob !
+```
+but will fail to parse:
+```
+person Bob
+Hello Sara !
+```
+because a `Person` object with the ID of 'Sara' has not been instantiated even though 'Sara' is a valid `ID`.
+
+#### Unordered Groups
+[NOTE] Unordered group are currently not supported but seems like it's being worked on. I think this is still valuable info that could be added to the documentation at a later stage if needed.
+
+By default, a parser rule has to be implemented in the exact order it is declared.
+```
+Person:
+    'person' name=ID age=INT
+```
+Here a `Person` object **needs** to first declare the property `name` then `age`.
+```
+person Bob 25
+```
+will successfully be parsed to an object of type `Person` while
+```
+person 25 Bob
+```
+will throw an error.
+
+However, it is possible to declare a group of properties in an unordered fashion using the `&` operator
+```
+Person:
+    'person' name=ID & age=INT
+```
+will now allow `name` and `age` to be declared in any order.
+```
+person 25 Bob
+```
+will then successfully create an object of type `Person`.
+
+Cardinality (?,*,+ operators) also applies to unordered group. Please note that assignments with a cardinality of `+` or `*` have to appear continuously and cannot be interrupted by an other assignment and resumed later.
+
+#### Simple Actions
+It is possible for a rule to return different types depending on declaration
+```
+RuleOne returns TypeOne:
+    'keywordOne' name=ID | RuleTwo;
+
+RuleTwo returns TypeTwo:
+    'keywordTwo' name=ID;
+```
+In the above example, we rely on a *rule call* to specify the return type. With more complex rules, the readability will be highly impacted. *Actions* allow to improve the readability of the grammar by explicitly defining the return type. *Actions* are declared inside of curly brackets `{}`:
+```
+RuleName returns TypeOne:
+    'keywordOne' name=ID | 'keywordTwo' {TypeTwo} name=ID;
+```
+
+#### Unassigned Rule Calls
+Parser rules do not necessarily need to return an object, they can also refer to other parser rules which in turn will be responsible for returning the object.
+For example, in the [Arithmetics example](https://github.com/langium/langium/blob/main/examples/arithmetics/src/language-server/arithmetics.langium):
+```
+AbstractDefinition:
+	Definition | DeclaredParameter;
+```
+The parser rule `AbstractDefinition` will not create an object of type AbstractDefinition. Instead, it calls either the `Definition` or `DeclaredParameter` parser rule which will be responsible for creating an object of a given type (or call other parser rules if they are unassigned rule calls themselves). 
+
+#### Assigned Actions
```

Comment:
```suggestion
#### Assignments
```

Reply:
There is already a section "Assignments" (line ~190) before where it is described how to assign data to a property/feature. 
Any idea on renaming that wouldn't be confusing with the previous section?